examples/some-verifier/src/main.rs

```diff
@@ -315,9 +256,103 @@ async fn add_verification(
 ) -> Result<StatusCode, Error> {
     tracing::info!("Request to verify.");
 
-    let Json(Request { proof, timestamp }) = request?;
+    let request = request?;
+    verify_request(&mut state, &request).await?;
+
+    let Json(Request { proof, .. }) = request;
+
+    // Check the statements and add them to the database
+    let num_statements = proof.verifiable_credential.len();
+    if num_statements < 2 {
+        return Err(Error::NotEnoughStatements(num_statements));
+    }
+
+    let mut entry = VerificationsEntry::from_presentation(&proof);
+    for proof in &proof.verifiable_credential {
+        proof_to_verifications_entry(state.clone(), proof, &mut entry)?;
+    }
+
+    match state.database.add_verification(entry).await {
+        Ok(()) => {}
+        // TODO: duplicate/overlapping entries
+        Err(err) => {
+            tracing::warn!("Error inserting entries: {err}");
+            return Err(Error::DatabaseError(err));
+        }
+    }
+
+    tracing::info!("Successfully verified.");
+
+    Ok(StatusCode::CREATED)
+}
+
+#[tracing::instrument(level = "info", skip_all)]
+async fn remove_verification(
+    State(mut state): State<AppState>,
+    request: Result<Json<Request>, JsonRejection>,
+) -> Result<StatusCode, Error> {
+    tracing::info!("Request to remove verification.");
+
+    let request = request?;
+    verify_request(&mut state, &request).await?;
+
+    let Json(Request { proof, .. }) = request;
+
+    // Check the statements and add them to the database
+    let num_creds = proof.verifiable_credential.len();
+    if num_creds != 1 {
+        return Err(Error::NotSingleStatement(num_creds));
+    }
+
+    let credential = proof.verifiable_credential.get(0).unwrap(); // We know we have exactly 1 credential
```

Comment:
```suggestion
    // We know we have exactly 1 credential
    let credential = proof.verifiable_credential.get(0).unwrap();
```
I prefer comments like this, but up to you